tianshou/data/collector.py

```diff
@@ -257,9 +283,16 @@ def collect(
                     f"n_step={n_step} is not a multiple of #env ({self.env_num}), "
                     "which may cause extra transitions collected into the buffer.",
                 )
+            if sample_equal_from_each_env:
+                warnings.warn("sample_equal_from_each_env is ignored when using n_step.")
             ready_env_ids = np.arange(self.env_num)
         elif n_episode is not None:
             assert n_episode > 0
+            if sample_equal_from_each_env:
```

Comment:
I generally dislike asserts, let's use ValueErrors instead. There is an option to run python programs such that asserts are ignored, so they should rather be viewed as "optional error checking" (even though many projects don't adhere to this).

Also, pls include input information in log messages, it makes them much more useful. Should be something like
```python
f"{n_episode=} must be a "
f"multiple of the number of envs when sample_equal_from_each_env is True but got {self.env_num=}."
```
```